src/optimizer/plugins/ConstructInlineInstanceOptimizer.ts

```diff
@@ -0,0 +1,69 @@
+import { pullAt, escapeRegExp } from 'lodash';
+import { OptimizerPlugin } from '../OptimizerPlugin';
+import { Package } from '../../processor';
+import { ExportableAssignmentRule, ExportableInstance } from '../../exportable';
+
+export default {
+  name: 'construct_inline_instance',
+  description:
+    'Construct inline instances from groups of rules in a contained resource or a Bundle',
+
+  optimize(pkg: Package): void {
+    const inlineInstances: ExportableInstance[] = [];
+    [...pkg.instances].forEach(instance => {
```

Comment:
I think your test cases provide good coverage of instances (including `Bundle`) that contain inline instances. However, any resource that extends `DomainResource` may contain inline instances in its `contained` property. So, that would include `StructureDefinition`, `CodeSystem`, and `ValueSet`.

Reply:
Oh oops! Good point. I've tested this out in SUSHI, and it looks like we don't do the necessary magic on ValueSets and CodeSystems to allow for inline instances being set on `contained`. So something like
```
ValueSet: MyVS
* ^contained = MyInlineObservation
```
doesn't appear to work in SUSHI (unless I'm missing something). But it does work on a `Profile`, so I will add support (and a test) for those.

Reply:
Added in https://github.com/FHIR/GoFSH/pull/45/commits/6b430f958bd9f32991dd01c625e5258b38f20567